service/storage_service.cc

```diff
@@ -6119,6 +6155,180 @@ future<> storage_service::stream_tablet(locator::global_tablet_id tablet) {
     }
 }
 
+future<join_node_request_result> storage_service::join_node_request_handler(join_node_request_params params) {
+    if (!_raft_topology_change_enabled) {
```

Comment:
May be just not register it in this case?

Reply:
If not it is better to return `disabled` result instead of throw.

Reply:
Now the verbs are not registered unless `_raft_topology_change_enabled` is true.